foam_controlwrapper/foam_mesh.py

```diff
@@ -0,0 +1,772 @@
+                if dataName not in dataNames:
+                    self.create_dummy_celldata(dataName,
+                                               [0, 0, 0, 0, 0, 0, 0])
+                foamFiles.set_cell_data(self.path, self.time,
+                                        dataName,
+                                        self._uuidToFoamLabel[cell.uid],
+                                        cell.data[data], 'scalar')
+            elif data == CUBA.VELOCITY:
+                dataName = "U"
+                if dataName not in dataNames:
+                    self.create_dummy_cellvectordata(dataName,
+                                                     [0, 1, -1, 0, 0, 0, 0])
+                foamFiles.set_cell_data(self.path, self.time,
+                                        dataName,
+                                        self._uuidToFoamLabel[cell.uid],
+                                        tuple(cell.data[data]), 'vector')
+            elif data == "":
+                pass
+            else:
+                error_str = "Data named "+data+" not supported"
+                raise NotImplementedError(error_str)
+
+    def iter_points(self, point_uuids=None):
+        """ Returns an iterator over the selected points.
+
+        Returns an iterator over the points with uuid in
+        point_ids. If none of the ids in point_ids exists,
+        an empty iterator is returned. If there is no ids
+        inside point_ids, a iterator over all points of
+        the mesh is returned instead.
+
+        Parameters
+        ----------
+        point_uuids : list of uuids, optional
+            uuids of the desired points, default empty
+
+        Returns
+        -------
+        iter
+            Iterator over the selected points
+
+        """
+
+        if point_uuids is None:
+            pointCount = foamface.getPointCount(self.name)
+            for label in range(pointCount):
+                point = self.get_point(self._foamPointLabelToUuid[label])
+                yield Point.from_point(point)
+        else:
+            for uid in point_uuids:
+                point = self.get_point(uid)
+                yield Point.from_point(point)
+
+    def iter_edges(self, edge_uuids=None):
+        """ Return empty list while edges are not supported yet
+
+        """
+
+        return []
+
+    def iter_faces(self, face_uuids=None):
+        """ Returns an iterator over the selected faces.
+
+        Returns an iterator over the faces with uuid in
+        face_uuids. If none of the uuids in face_uuids exists,
+        an empty iterator is returned. If there is no uuids
+        inside face_uuids, a iterator over all faces of
+        the mesh is returned instead.
+
+        Parameters
+        ----------
+        face_uuids : list of uuids, optional
+            Uuids of the desired faces, default empty
+
+        Returns
+        -------
+        iter
+            Iterator over the selected faces
+
+        """
+
+# create patch data
```

Comment:
style: There are a lot of comments that look out of place because of wrong indentation